### ISSUE TITLE
Fail to handle errors before init has ended

### ISSUE THREAD

**Issue by maboiteaspam** [slimphp/Slim#148]
Hi !

I was playing with Slim, and connected it to various other components.

I faced a big list of not correctly handled errors / exceptions.
My copy/paste is below.

Simply, i included Slim, throwed an exception, and then launched init.
And it fails.

I did not try to understand the why and how of errors, but from what i did understand, Slim has failed because, it is setting errors handlers in top of the included file.
Wich means before init is ok.
But, with an unfinished Init phase, it seems that the framework is not able to handle errors, without throwing errors himself.

By the way this is annoying, frustrating, because as you will see, errors reported does not concern throwed exception and is absolutely not understandable.
`There is an error, but i don't understand why, and where it come from ?!`

The solution would be to register errors handlers on init only, when Slim is ready to do so.

``` php
require($vendors_dir."Slim-1.3.5/Slim/Slim.php");
throw new Exception("sdfdsf");
// Init Slim
Slim::init( array(
    'view' => TwigView::init($app_dir."views/", $tmp_dir."tpl-cache/"),
    )
);
```

```
Notice: Trying to get property of non-object in /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php on line 438

Call Stack:
    0.0018     808048   1. Slim::handleExceptions() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0018     811292   2. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:198
    0.0018     812200   3. Slim::handleErrors() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0019     812276   4. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:178


Warning: array_keys() expects parameter 1 to be array, null given in /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php on line 438

Call Stack:
    0.0018     808048   1. Slim::handleExceptions() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0018     811292   2. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:198
    0.0018     812200   3. Slim::handleErrors() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0019     812276   4. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:178
    0.0128     812408   5. array_keys() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:438


Warning: in_array() expects parameter 2 to be array, null given in /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php on line 438

Call Stack:
    0.0018     808048   1. Slim::handleExceptions() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0018     811292   2. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:198
    0.0018     812200   3. Slim::handleErrors() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0019     812276   4. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:178
    0.0130     812408   5. in_array() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:438


Notice: Trying to get property of non-object in /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php on line 626

Call Stack:
    0.0018     808048   1. Slim::handleExceptions() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0018     811292   2. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:198
    0.0018     812200   3. Slim::handleErrors() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0131     812200   4. Slim::error() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:181
    0.0131     853632   5. Slim::router() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:595


Fatal error: Call to a member function error() on a non-object in /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php on line 595

Call Stack:
    0.0018     808048   1. Slim::handleExceptions() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0018     811292   2. Slim::config() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:198
    0.0018     812200   3. Slim::handleErrors() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:0
    0.0131     812200   4. Slim::error() /home/clement/Documents/www/slim-pizza/vendors/Slim-1.3.5/Slim/Slim.php:181
```

bye

**Comment by codeguy** [comment]
Yes this can certainly be improved. Thank you for your honest feedback. I'll work on improving this in the next point release.

**Comment by codeguy** [comment]
This is resolved as of 1.5.0.1. Hotfix pushed to both stable and develop branches. Slim will only use its custom error handler once the application has been fully instantiated. Otherwise, errors/exceptions are propagated to the existing/default error/exception handler.
